[Bug] Conan exports test executable bolt_shuffle_spark_memory_tests as a library target

### Component Selection

- [ ] Core Engine (Expression eval, Memory, Vector)
- [ ] Connectors / File Formats (Hive, Parquet, etc.)
- [ ] API / Bindings (Python, etc.)
- [x] Build
- [ ] Other

### Describe the Bug

```
CMake Error at build/Release/generators/cmakedeps_macros.cmake:66 (message):
  Library 'bolt_shuffle_spark_matrix_tests' not found in package.  If
  'bolt_shuffle_spark_matrix_tests' is a system library, declare it with
  'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/Release/generators/bolt-Target-release.cmake:23 (conan_package_library_targets)
  build/Release/generators/boltTargets.cmake:24 (include)
  build/Release/generators/bolt-config.cmake:16 (include)
  core/bolt.CMakeLists.cmake:143 (find_package)
  core/CMakeLists.txt:17 (include)


-- Configuring incomplete, errors occurred!
make[1]: *** [Makefile:90: build] Error 1
make[1]: Leaving directory '/data00/root/oss/gluten'
make: *** [Makefile:132: release-with-tests] Error 2
```

### Reproduction Steps

run `make cpp-test-releases` in gluten.

### Bolt Version / Commit ID

cf50e5c

### System Configuration

```markdown
### 1. Hardware & OS (Performance Context)
- **OS**: GNU/Linux 5.4.143.bsk.8-amd64 [Ubuntu 20.04.5 LTS]
- **Arch**: x86_64
- **CPU**: INTEL(R) XEON(R) PLATINUM 8582C
- **Cores**: 64
- **RAM**: 247 GB

### 2. Compiler Toolchain (Build Context)
- **C Compiler**: cc
  - Version: `cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0`
- **CXX Compiler**: g++
  - Version: `g++ (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0`
- **CMake**: 3.25.0
- **Ninja**:

### 3. Conan Dependency Manager
- **Conan Version**: Conan version 2.15.0
- **Conan Profile (default)**:

[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux

[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux
```

### Logs / Stack Trace

```shell

```

### Expected Behavior

_No response_

### Additional context

_No response_

--- a/bolt/shuffle/sparksql/tests/CMakeLists.txt
+++ b/bolt/shuffle/sparksql/tests/CMakeLists.txt
@@ -28,13 +28,13 @@ target_link_libraries(
 )
 
 add_executable(
-    bolt_shuffle_spark_matrix_tests
+    bolt_shuffle_spark_matrix_test
     ShuffleTestBase.cpp
     ShuffleMatrixTest.cpp
 )
 
 target_link_libraries(
-    bolt_shuffle_spark_matrix_tests
+    bolt_shuffle_spark_matrix_test
     PRIVATE
         bolt_shuffle_spark_impl
         bolt_exec_test_lib
@@ -47,18 +47,18 @@ target_link_libraries(
 
 # Add test
 add_test(
-    NAME bolt_shuffle_spark_matrix_tests
-    COMMAND bolt_shuffle_spark_matrix_tests
+    NAME bolt_shuffle_spark_matrix_test
+    COMMAND bolt_shuffle_spark_matrix_test
 )
 
 add_executable(
-    bolt_shuffle_spark_large_partition_tests
+    bolt_shuffle_spark_large_partition_test
     ShuffleTestBase.cpp
     ShuffleLargePartitionTest.cpp
 )
 
 target_link_libraries(
-    bolt_shuffle_spark_large_partition_tests
+    bolt_shuffle_spark_large_partition_test
     PRIVATE
         bolt_shuffle_spark_impl
         bolt_exec_test_lib
@@ -71,18 +71,18 @@ target_link_libraries(
 
 # Add test
 add_test(
-    NAME bolt_shuffle_spark_large_partition_tests
-    COMMAND bolt_shuffle_spark_large_partition_tests
+    NAME bolt_shuffle_spark_large_partition_test
+    COMMAND bolt_shuffle_spark_large_partition_test
 )
 
 add_executable(
-    bolt_shuffle_spark_memory_tests
+    bolt_shuffle_spark_memory_test
     ShuffleTestBase.cpp
     ShuffleMemoryTest.cpp
 )
 
 target_link_libraries(
-    bolt_shuffle_spark_memory_tests
+    bolt_shuffle_spark_memory_test
     PRIVATE
         bolt_shuffle_spark_impl
         bolt_exec_test_lib
@@ -94,6 +94,6 @@ target_link_libraries(
 
 # Add test
 add_test(
-    NAME bolt_shuffle_spark_memory_tests
-    COMMAND bolt_shuffle_spark_memory_tests
+    NAME bolt_shuffle_spark_memory_test
+    COMMAND bolt_shuffle_spark_memory_test
 )

